src/libsyntax/parse/token.rs

```diff
@@ -280,6 +280,14 @@ impl Token {
         }
     }
 
+    /// Returns `true` if the token can appear at the start of a const param.
+    pub fn can_begin_const_arg(&self) -> bool {
+        match self {
+            OpenDelim(Brace) => true,
+            _ => self.can_begin_literal_or_bool(),
```

Comment:
There are couple more non-controversial additions to this list - `NtExpr`, `NtBlock`.